### PR TITLE
feat(workflow): status/update ignore deleted outputs

### LINKED ISSUE #2802: Removed files still shown as outdated
**Describe the bug**
I removed some files, but renku status still sees these as outdated outputs. 

**Link to project**
https://renkulab.io/projects/remko.nijzink/budyko

**To Reproduce**
`renku status`
```
Outdated outputs(3):
  (use `renku update --all` to generate the file from its latest inputs)

	data/img/technical_paper/3_uncertainty.png: src_py/plot_uncertainty.py
	data/img/technical_paper/4_dist_aridity.png: src_py/plot_dist_aridity.py
	data/img_tech_note/4_dist_env.png: src_py/plot_dist_env.py

Modified inputs(3):

	src_py/plot_dist_aridity.py
	src_py/plot_dist_env.py
	src_py/plot_uncertainty.py

Deleted files used to generate outputs:

	src_flexsimple


```

**Expected behavior**
The files data/img/technical_paper/3_uncertainty.png and data/img/technical_paper/4_dist_aridity.png do not exist anymore and should not be shown.

**Run environment (please complete the following information):**
 - renku-python version 1.1.4
 - Python version 3.8.5
 - local terminal

### LINKED ISSUE #2843: CLI improvements
- `renku status` should work in dirty repositories since it doesn't modify anything
- `renku workflow visualize` should work for deleted files. This is useful to check deleted files that are reported by `renku status`.

### PR DESCRIPTION
**Pull request by mohammad-sdsc**
# Description

- Adds a flag to `status` and `update` for ignoring deleted generations.
- Fixes a bug where calling `status`/`update`/`workflow visualize` with paths from a subdirectory causes a crash
- Refactors `status` and `update` code to be more consistent
- Makes `status` to ignore deleted plans similar to `update`
- Some minor CLI improvements
- Fixes `runner` fixture to correctly separate `stdout` and `stderr`

Fixes #2802 
Fixes #2843

# Test
```
renku clone https://dev.renku.ch/gitlab/mohammad.alisafaee/delete-me-test-update.git

renku status
renku status --ignore-deleted
renku update --all --ignore-deleted
```

# TODO

- [x] The update/status logic should be moved from command and utility modules to the core module. This should be done once the review is done since there would be not change to the logic itself and it makes the review easier.

### CODE REVIEW

File: renku/command/status.py [hunk 1]
```diff
@@ -18,95 +18,76 @@
...
-            start_activity: Root activity to start from.
-
-        Returns:
-            Root activity and all its downstream activities.
-        """
-        all_activities = activity_gateway.get_downstream_activities(start_activity)
-        all_activities.add(start_activity)
-        return all_activities
-
+def _get_status(ignore_deleted, client_dispatcher: IClientDispatcher, activity_gateway: IActivityGateway, paths=None):
```
Comment (Panaetius) [inline renku/command/status.py:42]:
```suggestion
def _get_status(ignore_deleted: bool, client_dispatcher: IClientDispatcher, activity_gateway: IActivityGateway, paths=None):
```

File: renku/command/update.py [hunk 2]
```diff
@@ -46,6 +47,7 @@ def update_command():
 def _update(
     update_all,
     dry_run,
+    ignore_deleted,
```
Comment (Panaetius) [inline renku/command/update.py:50]:
```suggestion
    update_all: bool,
    dry_run: bool,
    ignore_deleted: bool,
```

File: renku/command/update.py [hunk 3]
```diff
@@ -117,35 +121,44 @@ def _is_activity_valid(activity: Activity, plan_gateway: IPlanGateway, client_di
     return plan.invalidated_at is None
 
 
-def _get_modified_activities_and_paths(repository, activity_gateway) -> Tuple[Set[Activity], Set[str]]:
-    """Return latest activities that one of their inputs is modified.
+def get_all_modified_and_deleted_activities_and_entities(
+    repository, activity_gateway
+) -> Tuple[Set[Tuple[Activity, Entity]], Set[Tuple[Activity, Entity]]]:
+    """
+    Return latest activities that one of their inputs is modified/deleted along with the modified/deleted input entity.
```
Comment (Panaetius) [inline renku/command/update.py:128]:
```suggestion
    Return latest activities with at least one modified or deleted input along with the modified/deleted input entity.
```
Current formulation sounds a bit off

File: renku/ui/cli/status.py [hunk 4]
```diff
@@ -56,13 +56,16 @@
 
 @click.command()
 @click.pass_context
+@click.option("--ignore-deleted", is_flag=True, help="Ignore deleted paths.")
```
Comment (Panaetius) [inline renku/ui/cli/status.py:59]: Should we add a shorthand like `-i` or `-d` ?

File: renku/ui/cli/update.py [hunk 5]
```diff
@@ -125,6 +125,21 @@
    $ renku update C
    $ renku update B C D
 
+Ignoring deleted paths
+~~~~~~~~~~~~~~~~~~~~~~
+
+The update command will regenerate any deleted files/directories. If you don't
+want to regenerate deleted paths, pass ``--ignore-deleted`` to the update
+command. You can make this as the default behavior by setting
```
Comment (Panaetius) [inline renku/ui/cli/update.py:133]:
```suggestion
command. You can make this the default behavior by setting
```

File: renku/ui/cli/update.py [hunk 6]
```diff
@@ -151,7 +166,8 @@
 @click.option(
     "config", "-c", "--config", metavar="<config file>", help="YAML file containing configuration for the provider."
 )
-def update(update_all, dry_run, paths, provider, config):
+@click.option("--ignore-deleted", is_flag=True, help="Ignore deleted paths.")
```
Comment (Panaetius) [inline renku/ui/cli/update.py:169]: also add a shorthand?

**Review by Panaetius — CHANGES_REQUESTED** [review]
Really nice! I only found some minor spelling issues

File: renku/ui/cli/status.py (hunk 4 above)
Comment (mohammad-sdsc) [reply renku/ui/cli/status.py:59]: I've added `-i`

**Review by Panaetius — APPROVED** [review]

**Review by Panaetius — APPROVED** [review]

**Pull request merged**